go/apps/dialogue/vumi_app.js

```diff
@@ -166,6 +167,27 @@ var DialogueApp = App.extend(function(self) {
     self.states.add('states:start', function() {
         return self.states.create(self.poll.start_state.uuid);
     });
+
+    self.types.httpjson = function(desc) {
+        self.http = new JsonApi(self.im);
+
+        var payload = {
+	        user: {
+		        answers: self.im.user.answers
+	        },
+	        contact: self.contact,
+            conversation_key: self.poll.conversation_key
+	    };
+
+	    return self
+	        .http.request(desc.method, desc.url, desc.method !== 'GET' ? { data: JSON.stringify(payload) } : {})
```

Comment:
For the non-`GET` cases, we should give `.http.request` the unstringified data. For the `GET` case, we should give it `null` to tell it to not use a body for the request.